velox/connectors/hive/HiveDataSource.cpp

```diff
@@ -207,7 +207,7 @@ std::optional<RowVectorPtr> HiveDataSource::next(
     resetSplit();
     return nullptr;
   }
-
+  facebook::velox::core::inputFileName = split_->getFileName();
```

Comment:
hey @Yuhta  spark use samilar way https://github.com/apache/spark/blame/4f7f1c436205630ab77d3758d7210cc1a2f0d04a/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala#L113C10-L113C10

and https://github.com/apache/spark/blame/4f7f1c436205630ab77d3758d7210cc1a2f0d04a/core/src/main/scala/org/apache/spark/rdd/InputFileNameHolder.scala#L37C27-L37C27

Reply:
Unfortunately this does not work with Velox task threading model.  You should do these instead:
1.  Add a field in `DriverCtx` for the file name
2. Set the file name in table scan operator
3. Add a reference to `DriverCtx` in `ExecCtx`
4. Get the file name from function through `EvalCtx->ExecCtx->DriverCtx`

Reply:
oh @Yuhta , currently for spark scenario with Gluten , we always make sure the task run under supportsSingleThreadedExecution, checking here https://github.com/apache/incubator-gluten/blob/154f8750e1f9cd93c4b5362bbca2db219193e01f/cpp/velox/compute/WholeStageResultIterator.cc#L131C1-L133C4

So this way actually works for spark scenario, does you want me to refactor to make it more general ?

Reply:
Yes please use `DriverCtx` as this can be potentially used by other engines

Reply:
hey @Yuhta there is a circular reference problem if we add a reference to `DriverCtx` in `ExecCtx`, since `ExecCtx` belongs to `velox_core` module but `DriverCtx` belongs to `velox_exec` module which link to `velox_core`, if we add `DriverCtx` ref in `ExecCtx`, then `velox_core` would also need depends on `velox_exec`.

Do you have suggestion how to avoid that or whether it is ok we just put a new `inputFileName` field directly in `ExecCtx` ?

Reply:
`ExecCtx` is created per operator and you cannot access the one you created in `TableScan` from a `FilterProject` function eval.  You can also try adding `DriverCtx` directly to `EvalCtx` in this case.